aten/src/ATen/native/native_functions.yaml

```diff
@@ -4524,6 +4524,17 @@
     CPU: _cholesky_helper_cpu
     CUDA: _cholesky_helper_cuda
 
+- func: cholesky_mod(Tensor self, bool upper=False) -> (Tensor result, Tensor err)
```

Comment:
Perhaps something worth discussing. Should we make this a separate function, or add an argument `jitter` (boolean) to the original function, which then dispatches internally to either variant.

Reply:
I think the jitter arg would make sense. But we should distinguish between a function that allows to specify a jitter value and the one here that automatically increases jitter until the decomposition succeeds.

Reply:
Could there be modes? For instance: None, 'auto', jitter_val, or something like that.

Reply:
Yeah I think that's reasonable. The python interface could look sth like

```
def cholesky(A: Tensor: upper: bool=False, jitter: Optional[Union[str, float]]='none') -> ...
```
where `jitter='none'` is the current behavior, `jitter='auto'` is the behavior in this PR, and `jitter=value` uses the specified jitter value. This is quite a bit of arg overloading, but could be ok. The one thing this wouldn't allow is to have the behavior in this PR starting from a provided value.

Reply:
I am not sure how to do this in the `native_functions.yaml` interface definition. It seems the best I can do is an optional Tensor. We could write a python-only wrapper, but then the interface in the C++ code would differ.